ruma-federation-api/src/event/get_room_state_ids/v1.rs

```diff
@@ -0,0 +1,52 @@
+//! [GET /_matrix/federation/v1/state_ids/{roomId}](https://matrix.org/docs/spec/server_server/r0.1.4#get-matrix-federation-v1-state-ids-roomid)
+
+use ruma_api::ruma_api;
+use ruma_identifiers::{EventId, RoomId};
+
+ruma_api! {
+    metadata: {
+        description: "Retrieves a snapshot of a room's state at a given event, in the form of event IDs",
+        method: GET,
+        name: "get_room_state_ids",
+        path: "/_matrix/federation/v1/state_ids/:room_id",
+        rate_limited: false,
+        requires_authentication: true,
+    }
+
+    #[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+    request: {
+        /// The room ID to get state for.
+        #[ruma_api(path)]
+        pub room_id: &'a RoomId,
+
+        /// An event ID in the room to retrieve the state at.
+        #[ruma_api(query)]
+        pub event_id: &'a EventId,
+    }
+
+    #[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+    response: {
+        /// The full set of authorization events that make up the state of the
+        /// room, and their authorization events, recursively.
+        pub auth_chain_ids: Vec<EventId>,
+
+
+        /// The fully resolved state of the room at the given event.
+        pub pdu_ids: Vec<EventId>,
+    }
+}
+
+impl<'a> Request<'a> {
+    /// Creates a new `Request` with the given event id and room id.
+    pub fn new(event_id: &'a EventId, room_id: &'a RoomId) -> Self {
+        Self { event_id, room_id }
+    }
+}
+
+impl Response {
+    /// Creates a new `Response` with the given auth chain ids and room state
+    /// ids.
```

Comment:
We usually wrap comments at the 100 chars like the code (although this is very inconsistent at the moment since rustfmt doesn't enforce it – see also #221.

```suggestion
    /// Creates a new `Response` with the given auth chain IDs and room state IDs.
```